app/src/main/kotlin/xyz/duckee/android/MainActivity.kt

```diff
@@ -118,51 +115,55 @@ class MainActivity : ComponentActivity() {
 
     override fun onResume() {
         super.onResume()
+        handleAirbridgeDeeplink()
+    }
 
-        Firebase.dynamicLinks
-            .getDynamicLink(intent)
-            .addOnSuccessListener(this) { pendingDynamicLinkData: PendingDynamicLinkData? ->
-                pendingDynamicLinkData?.link?.let { deepLink ->
-                    val path = deepLink.path ?: return@let
-                    when {
-                        path.startsWith("/recipe/") -> {
-                            val recipePattern = "/recipe/(\\d+)".toRegex()
-                            val matchResult = recipePattern.find(path)
-                            matchResult?.let { result ->
-                                val recipeId = result.groupValues[1]
-                                navigationController?.navigateToRecipeScreen(recipeId)
-                            }
-                        }
-
-                        path.startsWith("/detail/") -> {
-                            val detailPattern = "/detail/([\\w-]+)".toRegex()
-                            val matchResult = detailPattern.find(path)
-                            matchResult?.let { result ->
-                                val detailId = result.groupValues[1]
-                                navigationController?.navigateToDetailScreen(detailId)
-                            }
-                        }
-
-                        path == "/explore" -> {
-                            navigationController?.navigateToExploreTab()
-                        }
-
-                        path == "/collection" -> {
-                            navigationController?.navigateToCollectionTab()
-                        }
-
-                        path == "/signin" -> {
-                            navigationController?.navigateToSignInScreen()
-                        }
-
-                        else -> {
-                            Timber.tag("[DuckeeMainActivity]").w("Unhandled deep link path: $path")
-                        }
-                    }
+    private fun handleAirbridgeDeeplink() {
+        Airbridge.handleDeeplink(intent) { deeplink ->
+            handleDeeplink(deeplink)
+        }
+
+        Airbridge.handleDeferredDeeplink { deeplink ->
+            deeplink?.let { handleDeeplink(it) }
+        }
+    }
```

Comment:
Implemented Airbridge deeplink handling logic. This code is called when the app is launched or resumed via a deeplink, and navigates the user to the appropriate screen. The navigation controller is used to move to the screen corresponding to the deeplink URL.